.github/workflows/refresh-wordpress-major-and-beta.yml

```diff
@@ -42,19 +42,29 @@ jobs:
               shell: bash
               env:
                   FORCE_REBUILD: ${{ github.event.inputs.force_rebuild }}
-              run: PATH="${PATH}:${HOME}/.bun/bin" npx nx bundle-wordpress:major-and-beta playground-wordpress-builds
+              run: |
+                  PATH="${PATH}:${HOME}/.bun/bin" npx nx bundle-wordpress:major-and-beta playground-wordpress-builds
+                  if [ $? -eq 0 ]; then
+                    echo "WordPress build succeeded"
+                    echo 'BUILD_FAILED=0' >> $GITHUB_OUTPUT
+                  else
+                    echo "WordPress build failed"
+                    echo 'BUILD_FAILED=1' >> $GITHUB_OUTPUT
+                  fi
             - name: Check for uncommitted changes
+              # Prevent commit if the build failed to avoid polluting the commit history
+              if: steps.changes.outputs.BUILD_FAILED == '0'
```

Comment:
Is this check correct, given that `changes` is the ID of this step?

Reply:
Also, could this use the step outcome GitHub already exposes instead of introducing an additional variable?

https://github.com/actions/toolkit/issues/1034#issuecomment-1140492001

Reply:
Thanks! Using outcome is better [117725b](https://github.com/WordPress/wordpress-playground/pull/1955/commits/117725b11f59c72611687de3f36b153e540323dd).

Reply:
I wonder why did it work in the test runs if the condition was off

Reply:
Interesting. You [must have an id to set a output value.  ](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-output-parameter)

If there is no id the variable doesn't exist, but the condition `if: steps.changes.outputs.BUILD_FAILED == '0'` will be true.

You can see how [this example](https://github.com/bgrgicak/test-github-action-outputs/blob/main/.github/workflows/action.yml) action [runs in a test](https://github.com/bgrgicak/test-github-action-outputs/actions/runs/11592008203/job/32272970903). 

The lesson is. If you evaluate strings use something that isn't _0_ or _1_ as the string.